gulpfile.babel.js

```diff
@@ -128,12 +143,87 @@ const buildConfig = (invocationArgs, sourceRoot, testRoot, buildRoot) => {
 
 const config = buildConfig(process.argv, 'source', 'test', 'build');
 
+const httpCache = new Keyv({
+  store: new KeyvFile({
+    filename: `${config.dir.build.cache}/http.json`,
+  }),
+});
+
 // Shared config tasks
 
 const cleanSharedConfig = () => del(config.files.src.derivedConfigs);
 
 export const distributeSharedConfig = gulp.series(cleanSharedConfig, distributeConfig);
 
+// Font tasks
+
+const cleanFonts = () => del(config.files.src.fonts.concat([config.dir.src.sassFonts]));
+
+const compileFonts = () => {
+  fs.ensureDirSync(config.dir.src.fonts);
+
+  const fonts = yaml.safeLoad(fs.readFileSync(config.files.src.fontsDefinition));
+
+  const files = fonts.reduce(
+    (carry, font) => {
+      font.files.forEach(file => {
+        const uri = url.resolve(font.base, file.path);
+        if (!(uri in carry)) {
+          carry[uri] = [];
+        }
+        carry[uri].push({
+          fontDisplay: 'fallback',
+          fontFamily: font.name,
+          fontStyle: file.style || 'normal',
+          fontWeight: file.weight || 400,
+          subsetMap: font.subsets,
+        });
+      });
+
+      return carry;
+    }, {},
+  );
+
+  const fontCache = `${config.dir.build.cache}/fonts`;
+  fs.removeSync(fontCache);
+
+  return Promise.all(Object.keys(files).map(uri =>
+    download(uri, {cache: httpCache})
+      .then(data => tempWrite(data, path.basename(uri)))
+      .then(fontFile => {
+        return Promise.all(files[uri].map(file => {
+          file.fontFile = fontFile;
+          file.outputFolder = fontCache;
+          return fontRanger(file);
+        }))
+          .finally(() => fs.promises.unlink(fontFile));
+      })))
+    .then(() => copy(fontCache, config.dir.src.sassFonts, {
+        filter: '*.css',
+        rename: basename => path.basename(basename, '.css') + `.scss`,
+        transform: () => replaceStream('url(\'', 'url(\'#{$fonts-path}/'),
+      },
+    ))
+    .then(() => copy(fontCache, config.dir.src.fonts, {filter: '*.woff2'}));
+};
+
+const compileFontLicenses = () => {
+  fs.ensureDirSync(config.dir.src.fonts);
+
+  const fonts = yaml.safeLoad(fs.readFileSync(config.files.src.fontsDefinition));
```

Comment:
This happens in `compileFonts` as well. Worth refactoring so the yaml file is only loaded once per run?

Reply:
Can't see an obvious way to do that to cope with the file potentially changing...

Reply:
Hmm... would probably be more complex than just having the double load, so happy to leave this as is.